drivers/bridge/bridge.go

```diff
@@ -431,10 +428,21 @@ func (d *driver) Config(option map[string]interface{}) error {
 		d.config = config
 	} else {
 		config = &configuration{}
+		d.config = config
 	}
 
 	if config.EnableIPForwarding {
-		return setupIPForwarding(config)
+		err := setupIPForwarding(config)
+		if err != nil {
+			return err
+		}
+	}
+
+	if config.EnableIPTables {
+		natchain, filterchain, err := setupIPChains(config)
+		d.natchain = natchain
```

Comment:
If you want to separate generation of chains from assignment, then only assign if err == nil.
But probably we do not need that separation as the returned error will make the driver config fail. So I think we can assign to d.natchain and d.filterchain on the same line.